src/learn/network/concensus.md

```diff
@@ -1 +1,100 @@
+
+Learn more about **validators** at [here](https://aleo.org/faq/). (TODO: redirect to relative docs)  
+
+## AleoBFT
+AleoBFT is a new hybrid architecture for consensus. It is a DAG-based BFT protocol inspired by Narwhal and Bullshark. It incentivises validators to preserve network liveness and provers to scale proving capacity for Aleo ecosystem.
+
+AleoBFT guarantees instant finality once validators achieve consensus for each block. With instant finality, not only validators enjoy better node stability but also creates smooth experience for applications developers and users. And this guarantee makes interoperability with other ecosystem simpler.
+
+AleoBFT provers are computing core components of ZK proofs and receive shares of coinbase rewards by solving and producing these coinbase proofs, which is called Proof of Succint Work (PoSW). This incentivise provers to also become a validator themselves by accumulate and stake 10 million ACs. By having broader rewards distribution, it helps Aleo Network to achieve greater proving capacity, further decentralise and scaling Aleo network and fortifies censorship-resistence guarantee.
+
+## Bullshark and Narwhal
+
+### Bullshark
+Bullshark is a DAG(Directed Acyclic Graph)-based BFT (Byzantine Fault Tolerance) protocol, it separate the network communication layer from the ordering (consensus) logic. Each message contains a set of transactions, and a set of references to previous messages. Together, all the messages form a DAG that keeps growing – a message is a vertex and its references are edges. A vertex can be a proposal and an edge can be a vote.
+
+While different parties may see slightly different DAGs at any point in time due to the asynchronous nature of the network, each validator can still totally (fully) orders all the vertices (blocks) without sending a single extra message by just looking at its local view of the DAG.
+
+The DAG being used here is round-based DAG, each vertex is associated with round number, each round has at most `n` vertices. Each validator broadcasts one message per round and each message references at least `n − f` messages in previous round. `n` is the total number of validating nodes in the network and `f` is the number of Byzantine nodes. Below shows a diagram of how it looks like with `n = 4` and `f = 1`.
+
+![DAG1](./images/DAG1.png)
+Diagram 1: Round-based DAG  
+Image from https://decentralizedthoughts.github.io/2022-06-28-DAG-meets-BFT/
+
+### Ordering Logic
+A predefined leader is elected in every even round and the vertex associated with it is referred as anchor. Anchor is a more suitable term to describe here because unlike typical leader-based protocols where the leader has to do all the works at each round and disseminate data to all other nodes, the anchor here only get chosen to commit its casual history once gather enough votes (`2f + 1`) (3 in this example).  
```

Comment:
Thanks for the suggestion! I didn't know that because haven't dig much into source code yet. Will do that next.

Reply:
@ghostant-1017 To confirm the fact, is `predefined leader` determined from future round like in the example you mentioned? I always thought `predefined` means defined before the round happens, where should I read in the source code to understand this more?

Reply:
I think there may be a confusion between a "Steady-State Leaders" and a "Fallback Leaders". At least according to the (Bullshark paper)[https://arxiv.org/abs/2201.05677], there are two types of leaders that get elected in different ways and in different rounds of each "wave":

1. Steady-State Leaders:
* These are predefined leaders in the first and third rounds of each wave.
* Steady-state leaders are committed during synchronous periods when network conditions are stable and predictable​​.

2. Fallback Leaders:
* These leaders are chosen retrospectively based on randomness generated in the fourth round of each wave.
* Fallback leaders are used during asynchronous periods when network conditions are less predictable, providing a robust mechanism to ensure progress despite network delays or faults​​.

In the last appendix of the paper referring to a "partially synchronous" version (which I think is the one used by Aleo), it reads:

> To totally order the vertices in the DAG, each party 𝑝 locally 𝑖
> interprets 𝐷𝐴𝐺 (there is no extracommunication on top of building 𝑖
> theDAG).To this end, 𝑝 divides its DAG into waves of 4 rounds 𝑖
> each. Every wave has 3 leaders that can potentially be committed:
> 2 steady-state leaders and one fallback leader. The steady-state
> leaders are two pre-defined vertices, one in the first round of the
> wave and the other in the third. The fallback leader is a vertex in the
> first round of the wave that is selected by the randomness produced
> in the fourth round of the wave. To make sure a fallback leader
> and a steady state leader are not committed in the same wave, each
> party can only vote for either the fallback leader or the steady-
> state ones. In the code, steadyVoters[w] fallbackVoters[w] contain
> all the parties that can vote for steady-state or fallback leaders in
> wave 𝑤, respectively.

Reply:
Thanks for the feedbacks! I believe Aleo implementation is a little different from what described in original whitepaper and too bad there's no clear reference for this from current resources other than codebase.

I think we can move forward first without extending the term `predefined leader` for now to avoid confusion and uncertainty on the actual fact before studying the source code.